### DISCUSSION TITLE
Is it possible to split database code into separate files?

### DISCUSSION THREAD

**Discussion by msarkrish** [simolus3/drift#2575] [Q&A] [answered]
Hi Simolus,

Thank you for this Plugin. 
In our application we are following feature based architecture. Database class is currently in the common folder. currently I'm planned to split the database class in to separate files based on the feature. 
DB initalization and table creation alone in common folder and all the query methods are written inside the each feature folder.

Is it possible Simolus?

**Comment by simolus3** [comment, marked as answer]
> DB initalization and table creation alone in common folder

So this one you can just keep, right? Since all your database code is in the common folder at the moment.

> all the query methods are written inside the each feature folder

Yes, you can do this with [DAOs](https://drift.simonbinder.eu/docs/advanced-features/daos/) that live in the different feature folders and just include the tables they need.

**Comment by msarkrish** [comment]
Thank you @simolus3 . You are genius.
